adapters/validation.py

```diff
@@ -0,0 +1,213 @@
+from .exceptions import CycleError, ValidationError
+
+
+class ValidationNode(object):
```

Comment:
Yes, I intend to target Python 3 only, but out of habit I always explicitly subclass from `object`. I really don't mind doing it either way. Maybe we should look into how Django will do it in 2.0 and then do the same?